### ISSUE TITLE
ignoreFrontMatter option doesn't work with some languages

### ISSUE THREAD

**Issue by fabon-f** [markuplint/markuplint#926]
- Markuplint version: `3.8.0`
- Parser lang: nunjucks, ejs, mustache and liquid
- Node.js version: `20.0.0`
- OS: macOS

## Describe the bug

Linting whole HTML file with `ignoreFrontMatter` raises error because of frontmatter in some languages. (FYI partial templates don't raise error.)

## Code Example

```html
<div>
...
</div>
```

## Steps To Reproduce

1. Add `.markuplintrc` and install plugins
```json
{
  "parser": {
    "\\.njk$": "@markuplint/nunjucks-parser",
    "\\.ejs$": "@markuplint/ejs-parser",
    "\\.mustache$": "@markuplint/mustache-parser",
    "\\.liquid": "@markuplint/liquid-parser"
  },
  "extends": [
    "markuplint:recommended"
  ],
  "parserOptions": {
    "ignoreFrontMatter": true
  }
}
```
2. Create `a.njk`, `a.ejs`, `a.mustache`, `a.liquid` with content below
```html
---
test: foobar
---

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Document</title>
</head>
<body>
    <h1>test</h1>
</body>
</html>
```
3. run `markuplint a.njk a.ejs a.mustache a.liquid`

## The current behavior

Linting raises many errors.

## The expected behavior

No errors should be found.

**Comment by YusukeHirao** [comment]
@fabon-f 

Thank you for your report.
I'll confirm the issue.
